tests/foreman/cli/test_usergroup.py

```diff
@@ -0,0 +1,510 @@
+# -*- encoding: utf-8 -*-
+"""Test class for User Group CLI"""
+
+from robottelo.cli.usergroup import UserGroup
+from robottelo.cli.base import CLIReturnCodeError
+from robottelo.cli.factory import (
+    CLIFactoryError,
+    make_role,
+    make_user,
+    make_usergroup,
+)
+from robottelo.datafactory import (
+    invalid_values_list,
+    valid_data_list,
+    valid_usernames_list,
+)
+from robottelo.decorators import tier1, tier2
+from robottelo.test import CLITestCase
+
+
+class UserGroupTestCase(CLITestCase):
+    """User group CLI related tests."""
+
+    @tier1
+    def test_positive_create_with_name(self):
+        """Create new user group using different valid names
+
+        @Feature: Usergroup
+
+        @Assert: User group is created successfully.
+        """
+        for name in valid_data_list():
+            with self.subTest(name):
+                user_group = make_usergroup({'name': name})
+                self.assertEqual(user_group['name'], name)
+
+    @tier1
+    def test_positive_create_with_user_name(self):
+        """Create new user group using valid user attached to that group. Use
+        user name as a parameter
+
+        @Feature: Usergroup
+
+        @Assert: User group is created successfully.
+        """
+        for login in valid_usernames_list():
+            with self.subTest(login):
+                user = make_user({'login': login})
+                user_group = make_usergroup({'users': user['login']})
+                self.assertEqual(user_group['users'][0], login)
+
+    @tier1
+    def test_positive_create_with_user_id(self):
+        """Create new user group using valid user attached to that group. Use
+        user id as a parameter
+
+        @Feature: Usergroup
+
+        @Assert: User group is created successfully.
+        """
+        user = make_user()
+        user_group = make_usergroup({'user-ids': user['id']})
+        self.assertEqual(user_group['users'][0], user['login'])
+
+    @tier1
+    def test_positive_create_with_users(self):
+        """Create new user group using multiple users attached to that group.
+        Use users name as a parameter
+
+        @Feature: Usergroup
+
+        @Assert: User group is created successfully and contains all expected
+        users.
+        """
+        users = [make_user()['login'] for _ in range(3)]
+        user_group = make_usergroup({'users': users})
+        for user in user_group['users']:
```

Comment:
I think a more strict test would be matching both lists `self.assertEqual(sorted(users), sorted(user_group['users'])`.

PS using sorted to make sure both lists have the same order to avoid false negatives like:

```python
>>> [1, 2] == [2, 1]
False
```

Another approach would be converting the lists to set, but it would remove duplicates which can mask if the server returns duplicated data.